ci/windows_dev_env/windows_deps_headless_installer.py

```diff
@@ -0,0 +1,429 @@
+    vs_file_path = vs_file_path + '.exe'
+    ret = call(vs_file_path +
+               ' --add Microsoft.VisualStudio.Workload.ManagedDesktop'
+               ' --add Microsoft.VisualStudio.Workload.NetCoreTools'
+               ' --add Microsoft.VisualStudio.Workload.NetWeb'
+               ' --add Microsoft.VisualStudio.Workload.Node'
+               ' --add Microsoft.VisualStudio.Workload.Office'
+               ' --add Microsoft.VisualStudio.Component.TypeScript.2.0'
+               ' --add Microsoft.VisualStudio.Component.TestTools.WebLoadTest'
+               ' --add Component.GitHub.VisualStudio'
+               ' --add Microsoft.VisualStudio.ComponentGroup.NativeDesktop.Core'
+               ' --add Microsoft.VisualStudio.Component.Static.Analysis.Tools'
+               ' --add Microsoft.VisualStudio.Component.VC.CMake.Project'
+               ' --add Microsoft.VisualStudio.Component.VC.140'
+               ' --add Microsoft.VisualStudio.Component.Windows10SDK.15063.Desktop'
+               ' --add Microsoft.VisualStudio.Component.Windows10SDK.15063.UWP'
+               ' --add Microsoft.VisualStudio.Component.Windows10SDK.15063.UWP.Native'
+               ' --add Microsoft.VisualStudio.ComponentGroup.Windows10SDK.15063'
+               ' --wait'
+               ' --passive'
+               ' --norestart'
+               )
+
+    if ret == 3010 or ret == 0:
+        # 3010 is restart required
+        logging.info("VS install successful.")
+    else:
+        raise RuntimeError("VS failed to install, exit status {}".format(ret))
+    # Workaround for --wait sometimes ignoring the subprocesses doing component installs
+
+    def vs_still_installing():
+        return {'vs_installer.exe', 'vs_installershell.exe', 'vs_setup_bootstrapper.exe'} & set(map(lambda process: process.name(), psutil.process_iter()))
+    timer = 0
+    while vs_still_installing() and timer < DEFAULT_SUBPROCESS_TIMEOUT:
+        logging.warning("VS installers still running for %d s", timer)
+        if timer % 60 == 0:
+            logging.info("Waiting for Visual Studio to install for the last {} seconds".format(str(timer)))
+        sleep(1)
+        timer += 1
+    if vs_still_installing():
+        logging.warning("VS install still running after timeout (%d)", DEFAULT_SUBPROCESS_TIMEOUT)
+    else:
+        logging.info("Visual studio install complete.")
+
+
+def install_cmake():
+    logging.info("Installing CMAKE")
+    cmake_file_path = download(DEPS['cmake'], '.')
+    check_call(['msiexec ', '/n', '/passive', '/i', cmake_file_path])
+    logging.info("CMAKE install complete")
+
+
+def install_perl():
+    logging.info("Installing Perl")
+    with tempfile.TemporaryDirectory() as tmpdir:
+        perl_file_path = download(DEPS['perl'], tmpdir)
+        check_call(['msiexec ', '/n', '/passive', '/i', perl_file_path])
+    logging.info("Perl install complete")
+
+
+def install_clang():
```

Comment:
Why do we need clang when compiling with visual studio?

Reply:
They probably need to support tvm op build on windows as well, which may need this.